test/MUXControlsTestApp/Utilities/TestHelpers.cs

```diff
@@ -126,7 +126,10 @@ public static void ClearVisualTreeRoot()
             {
                 FrameworkElement rootElement = MUXControlsTestApp.App.TestContentRoot as FrameworkElement;
 
-                if (rootElement != null)
+                if (rootElement != null
+                // In MUXControlsTestApp.App.TestContentRoot setter, we set the content to MainPage if value is null.
+                // Therefore UnloadedEvent will never be triggered if rootElement type is MainPage, no need to add event here.
+                && rootElement.GetType() != typeof(MainPage))
```

Comment:
>rootElement.GetType() != typeof(MainPage) [](start = 19, length = 41)

Can you use "!(rootElement is MainPage)"? "is" keyword is preferred over GetType() if you can.

And also why won't the UnloadedEvent fire in the case that the root is MainPage?

Reply:
When test started the default page is MainPage. Those 2 failed tests are skipped pre RS2, which means they return immediately and don't make any changes to the page content. During test cleanup we call ClearVisualTreeRoot to clear the rootElement, it'll set the content back to MainPage, but the content is already MainPage in this case, so UnloadedEvent never fires.